lib/github.ml

```diff
@@ -304,15 +304,65 @@ let check_tag ~dry_run vcs tag =
   if Vcs.tag_exists ~dry_run vcs tag then Ok ()
   else
     R.error_msgf
-      "CHANGES.md lists '%s' as the latest release, but no corresponding tag \
+      "CHANGES.md lists '%a' as the latest release, but no corresponding tag \
        has been found in the repository.@.Did you forget to call 'dune-release \
        tag' ?"
-      tag
+      Vcs.Tag.pp tag
 
 let assert_tag_exists ~dry_run tag =
   Vcs.get () >>= fun repo ->
   if Vcs.tag_exists ~dry_run repo tag then Ok ()
-  else R.error_msgf "%s is not a valid tag" tag
+  else R.error_msgf "%a is not a valid tag" Vcs.Tag.pp tag
+
+let is_annotated_tag vcs remote_rev =
```

Comment:
Personally I would like to come up with a better name for "unpeeled" in general because to me it is quite confusing to have negation in the name (I don't think there is a git term for it) so maybe "wrapped" or "outer" would make sense? It is even more annoying because at this point we don't know whether it is an "unpeeled"/"wrapped"/"outer" commit at all, that's exactly what we're trying to determine in this function. I would be very glad for a good name.

(But `remote_rev` is also wrong. I think in this function it should be called `commit_ish`, because the function is more general and checks exactly for that, even despite only being passed tags)